plugins/fastem_roa_import_export.py

```diff
@@ -0,0 +1,341 @@
+        ImportExportTool().show(
+            project_list_controller=self._acquisition_tab._project_list_controller
+        )
+
+
+class ImportExportTool(object):
+    """
+    The class ImportExportTool is a singleton that manages creating and showing
+    ImportExportFrame class.
+
+    """
+
+    # Note: This is the Borg design pattern which ensures that all
+    # instances of this class are actually using the same set of
+    # instance data.  See
+    # http://aspn.activestate.com/ASPN/Cookbook/Python/Recipe/66531
+    __shared_state = None
+
+    def __init__(self):
+        if not ImportExportTool.__shared_state:
+            ImportExportTool.__shared_state = self.__dict__
+        else:
+            self.__dict__ = ImportExportTool.__shared_state
+
+        if not hasattr(self, "initialized"):
+            self.initialized = False
+
+    def init(self, pos=wx.DefaultPosition, size=wx.Size(300, 100), app=None):
+        """
+        Init is used to set some parameters that will be used later
+        when the inspection tool is shown.  Suitable defaults will be
+        used for all of these parameters if they are not provided.
+
+        :param pos:   The default position to show the frame at
+        :param size:  The default size of the frame
+        :param app:  A reference to the :class:`App` object.
+        """
+        self._frame = None
+        self._pos = pos
+        self._size = size
+        self._app = app
+        if not self._app:
+            self._app = wx.GetApp()
+        self.initialized = True
+
+    def show(self, project_list_controller=None):
+        """
+        Creates the import export feature frame and raises it if neccessary.
+
+        :param pos: The default position to show the frame at.
+        :param size: The default size of the frame.
+        :param app: A reference to the class App object.
+        :param project_list_controller: (FastEMProjectListController) class which creates/removes new
+            FastEM projects.
+
+        """
+        if not self.initialized:
+            self.init()
+
+        parent = self._app.GetTopWindow()
+        if not self._frame:
+            self._frame = ImportExportFrame(
+                parent=parent,
+                pos=self._pos,
+                size=self._size,
+                project_list_controller=project_list_controller,
+            )
+
+        self._frame.Show()
+        if self._frame.IsIconized():
+            self._frame.Iconize(False)
+        self._frame.Raise()
+
+
+class ImportExportFrame(wx.Frame):
+    """
+    This class is the frame that holds the Import Export ROAs tool. The toolbar is managed here.
+
+    :param project_list_controller: (FastEMProjectListController) class which creates/removes new
+            FastEM projects.
+
+    """
+
+    def __init__(self, title="Import Export ROAs Tool", project_list_controller=None, *args, **kw):
+        kw["title"] = title
+        wx.Frame.__init__(self, *args, **kw)
```

Comment:
Typically it's best to write `super().__init__(*args, **kw)` (that works better with copy-paste, and that handles multiple inheritance).